.changelog/21616.txt

```diff
@@ -0,0 +1,6 @@
+```release-note: improvement
+connect: Add Envoy 1.31 and 1.30 to support matrix
+```
+```release-note: breaking-change
```

Comment:
Should we mention this breaking change in our documentation under `/website`?

Reply:
I believe we will need to update https://developer.hashicorp.com/consul/docs/connect/proxies/envoy#supported-versions when 1.20 is released, what I don't know is if we should backport this update to 1.19

Reply:
Moving our supported range on `main` for the next release doesn’t constitute a breaking change, so I think we can drop the second changelog entry - we document that we support a set number of Envoy versions in each release, so as long as there’s an overlap between the versions supported by Consul `n` and `n-2`, there’s an upgrade path available that matches our supported Consul upgrade path. We don’t typically remove backwards compatibility testing (what this range effectively determines) for Envoy versions in existing releases - customers may still be on an unmaintained Envoy version due to using an older version of Consul, so we don’t want to break existing compatibility in our backports unless forced to by a significant bug or CVE.

I’d stick with this bump on `main` for the 1.20 release, and leave older versions as they are (we’ll still be able to upgrade to maintained version patches when fixes come out). We could add 1.30 to Consul 1.19 as well so that the gap from 1.19 to 1.20 is back down to 1 Envoy version, but I think that’s a product call and optional. cc @blake